src/modules/rdf/functions.cpp

```diff
@@ -357,21 +357,24 @@ bool RDFModule::calculateGR(GenericList &processingData, ProcessPool &procPool,
 
     procPool.resetAccumulatedTime();
     timer.start();
-    for_each_pair(0, originalgr.nAtomTypes(), [&originalgr, &procPool, method](auto typeI, auto typeJ) {
-        // Sum histogram data from all processes (except if using RDFModule::TestMethod, where all processes
-        // have all data already)
-        if (method != RDFModule::TestMethod)
-        {
-            if (!originalgr.fullHistogram(typeI, typeJ).allSum(procPool))
-                return false;
-            if (!originalgr.boundHistogram(typeI, typeJ).allSum(procPool))
-                return false;
-        }
-
-        // Create unbound histogram from total and bound data
-        originalgr.unboundHistogram(typeI, typeJ) = originalgr.fullHistogram(typeI, typeJ);
-        originalgr.unboundHistogram(typeI, typeJ).add(originalgr.boundHistogram(typeI, typeJ), -1.0);
-    });
+    for_each_pair_early(0, originalgr.nAtomTypes(),
```

Comment:
```suggestion
    success = for_each_pair_early(0, originalgr.nAtomTypes(),
```
Then, after the loop
````
    if (!success) return false;
````

Otherwise, the double loop exits early, but the overall method still returns true.